src/server/game/Spells/SpellEffects.cpp

```diff
@@ -6058,7 +6058,12 @@ void Spell::SummonGuardian(uint32 i, uint32 entry, SummonPropertiesEntry const*
 
         // xinef: do not use precalculated position for effect summon pet in this function
         // it means it was cast by NPC and should have its position overridden
-        if (totalNumGuardians == 1 && GetSpellInfo()->Effects[i].Effect != SPELL_EFFECT_SUMMON_PET)
+        // unless the target is a position specified in the DB
+        if
+        (
+            (totalNumGuardians == 1 && GetSpellInfo()->Effects[i].Effect != SPELL_EFFECT_SUMMON_PET) ||
+            (!GetSpellInfo()->Effects[i].HasRadius() && GetSpellInfo()->Effects[i].TargetA.GetTarget() == TARGET_DEST_DB)
+        )
```

Comment:
```suggestion
        if ((totalNumGuardians == 1 && GetSpellInfo()->Effects[i].Effect != SPELL_EFFECT_SUMMON_PET) ||
            (!GetSpellInfo()->Effects[i].HasRadius() && GetSpellInfo()->Effects[i].TargetA.GetTarget() == TARGET_DEST_DB))
```

Reply:
I updated the format but added a new condition for a zero-radius spells. Trying to walk the line between compact and unreadable. I still prefer the original format (I think breaking it out is way easier to read), but let me know what you think of this hybrid approach.